### PR TITLE
[stdlib] Change `msg` argument in `assert_true`/`assert_false`/... to keyword-only

### PR DESCRIPTION
**Pull request by softmaxer**
changes:

- Add * in function definitions of stdlib/src/testing/testing.mojo to separate variadic and keyword only arguments.
- Scan for call sites of these assert functions and replace assert_true(val, msg) to assert_true(val, msg=msg) for both assert_true and assert_false

Normally, This should fix https://github.com/modularml/mojo/issues/2487 .
If any other changes are needed, Please let me know!

As mentioned by a colleague on discord:
```
The change is meant to remove a footgun caused by implicit conversion
```
Reference issue: #2475 

@jackos  @JoeLoser  As discussed in the Previous PR #2490  which was closed due to unnecessary commits, I am reopening the same PR again, this time I hope the same issues from the last PR don't exist.

 > This is failing one test in the file `test_memory.mojo` and I don't quite see why.
I think I figured it out, I accidentally deleted one line in a test function, sorry :p . I added it back now.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/doc_issue.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Documentation issue
 description: Report a problem with the Mojo docs
 title: "[Docs]"
 labels:
-  - documentation
```
Comment (laszlokindrat) [inline .github/ISSUE_TEMPLATE/doc_issue.yaml:5]: What are these changes for?

Comment (softmaxer) [reply .github/ISSUE_TEMPLATE/doc_issue.yaml:5]: Hey! so yeah this was a commit that was not intended. I tried reverting it but didn't work out. I will try again.

**Comment by softmaxer** [comment]
> Thanks for the patch! It would be great to have this, but I don't see the changes to the actual `assert_*` functions; did something go wrong during a rebase? Also, please make sure you mention this in the changelog!

Yes! I think something went wrong during a rebase, I'm gonna make a new commit!

File: .github/ISSUE_TEMPLATE/doc_issue.yaml (hunk 1 above)
Comment (laszlokindrat) [reply .github/ISSUE_TEMPLATE/doc_issue.yaml:5]: I'm not sure. @JoeLoser do you know how to help with this DCO issue?

File: docs/changelog.md [hunk 2]
```diff
@@ -592,3 +592,6 @@ by [@jayzhan211](https://github.com/jayzhan211))
   will also see large improvements.
 - [#2692](https://github.com/modularml/mojo/issues/2692) Fix `assert_raises`
   to include calling location.
+- [#2475](https://github.com/modularml/mojo/issues/2487) Fix `remove a 
+  footgun caused by implicit conversion by changing msg to keyword 
+  argument in assert_true/assert_false/ ..`
```
Comment (laszlokindrat) [inline docs/changelog.md:597]:
```suggestion
- [#2475](https://github.com/modularml/mojo/issues/2487) Remove a 
  footgun caused by implicit conversions in `assert_true`, `assert_false`, etc.
```

**Comment by laszlokindrat** [comment]
@softmaxer Could you please rebase on the latest nightly? I'm still seeing some diffs that don't seem relevant.

**Comment by laszlokindrat** [comment]
@softmaxer Do you think this PR is still useful? We have made several changes to implicit conversion rules recently that might mitigate problems.

**Comment by softmaxer** [comment]
> @softmaxer Do you think this PR is still useful? We have made several changes to implicit conversion rules recently that might mitigate problems.

@laszlokindrat I'm afraid you're right.
I couldn't address this issue any sooner due to some other work and later on I realized that this PR may no longer be appropriate.

Should I just close this now?

**Comment by laszlokindrat** [comment]
Thanks for working on this! We're looking forward to another contribution from you :)

**Pull request closed without merging**